### PR TITLE
Refactor Project

### LINKED ISSUE #55: [Feature] Add logging package
**Summarize the feature**
To easier determine errors, it would be handy if a logging package would be used. For that, I recommend [Loguru](https://github.com/Delgan/loguru).

**Example of the feature**
Loguru enables the developer to just import `from loguru import logger` and then use logging levels like **debug**, **info**, and **error**. In comparison to the default logger, Loguru logs are already nicely formatted.

**Additional context**
As an alternative, the built-in logging package could be used.

### LINKED ISSUE #56: [Feature] Add a package manager and `pyproject.toml`
**Summarize the feature**
The current setup with `setup.py` and all the other settings files is not quite convenient. Additionally, the packages within the `requierements.txt` are not correctly checked for compatibility. For that, I recommend using [Poetry](https://python-poetry.org/) or [Pipenv](https://pipenv.pypa.io/en/latest/index.html). 

Further, I recommend switching most of the configurations to the `pyproject.toml` ([docs](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/)). 

**Example of the feature**
Using Poetry, it is possible to easily manage packages using `poetry add` and `poetry install`. It is also possible to define different groups like **dev** or **docs** so that you can exclude not necessary packages.

**Additional context**
None

### LINKED ISSUE #57: [Feature] Switch from sphinx to mkdocs
**Summarize the feature**
In my opinion, Mkdocs is a more modern and easier approach to documentation in python.

Mkdocs is also compatible with [readthedocs](https://docs.readthedocs.io/en/stable/intro/getting-started-with-mkdocs.html).

**Example of the feature**
[FastAPI](https://fastapi.tiangolo.com/), [mkdocs-material](https://squidfunk.github.io/mkdocs-material/getting-started/)

**Additional context**
None

### PR DESCRIPTION
**Pull request by felixhoffmnn**
## Description

This PR includes potential **breaking changes**!

- Fixed #57 
- Fixed #56 
- Fixed #55 

## Overview

- Removed old project config files (python, sphinx / docs, and unit tests)
- Added config files (IDE, VS Code, pytest, codecov, formatting, linting, pre-commit, prettier, cspell, readthedocs, and poetry)
- Updated GitHub Actions CI pipeline
- Switched from sphinx to mkdocs
- Improved typing for all functions
- Added partial logging
- Refactored library functions
- Moved vvs data

## Test Plan

- The core functionality should not have changed
- But the test coverage is very low (To-dos need to be revisited)

## Checklist

- [x] Connected regarding _GitHub issues_ and _discussions_
- [x] Added _docstrings_ for all new functions
- [x] Checked _CI_ pipeline for any failures
- [x] Run _pytests_ locally to check for any failures

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,13 +1,10 @@
-* @zaanposni
+- @zaanposni
 
 # PYTHON
-*.py @zaanposni
-/requirements.txt @zaanposni
 
-# UNITTESTS
-/tests/ @zaanposni
+\*.py @zaanposni
```
Comment (felixhoffmnn) [inline .github/CODEOWNERS:5]: Needs to be fixed

File: docs/vvspy/functions.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# VVSPY Function Documentation
```
Comment (felixhoffmnn) [inline docs/vvspy/functions.md:1]: Add text

File: docs/vvspy/models.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# VVSPY Models
+
```
Comment (felixhoffmnn) [inline docs/vvspy/models.md:2]: Add text

File: docs/vvspy/utils.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Python Utilities
+
```
Comment (felixhoffmnn) [inline docs/vvspy/utils.md:2]: Add text

File: vvspy/trips.py [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+
+    Returns
+    -------
+    List[Trip]
+        A list of Departure objects or None if a error occurred.
+    """
+    parsed_response = []
+
+    try:
+        print(result["journeys"])
```
Comment (zaanposni) [inline vvspy/trips.py:33]: did you mean to add debugging here or did you forget to remove that print statement ^.^ ?

```suggestion
```

File: .github/CODEOWNERS (hunk 1 above)
Comment (zaanposni) [reply .github/CODEOWNERS:5]: We can add you as a codeowner too if you want to

**Review by zaanposni — CHANGES_REQUESTED** [review]

File: .github/workflows/pytest.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+name: Run Pytest
+
+on:
+  push:
+    branches: ["*"]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: ["main"]
```
Comment (zaanposni) [inline .github/workflows/pytest.yaml:8]: For now vvspy uses "master" as default branch. 
We can change that later on.
For now please change it here, so we can test your workflows.

Comment (felixhoffmnn) [reply .github/workflows/pytest.yaml:8]: Ah I see. Did not notice it in dev.

File: .github/CODEOWNERS (hunk 1 above)
Comment (felixhoffmnn) [reply .github/CODEOWNERS]: For now, I will only change the syntax.

**Pull request closed without merging**